### PR TITLE
fix: Safer fixtures

### LINKED ISSUE #31: Safer fixture
Looking at the code of the fixtures, I see that the yield operation is not protected

For example:

https://github.com/microsoft/playwright-pytest/blob/fb5a303999e018b75e006edc0ceb1e4f0b4c67be/pytest_playwright/pytest_playwright.py#L97-L98

I think it should be:

```
try:
    yield pw
finally:
    pw.close()
```

to ensure fixture are closed even if an error occurs within a test.

> This apply to other fixtures as well

### PR DESCRIPTION
**Pull request by kumaraditya303**
Fix #31 ,  and updated pre-commit hooks

### CODE REVIEW

File: pytest_playwright/pytest_playwright.py [hunk 1]
```diff
@@ -118,17 +121,21 @@ def launch(**kwargs: Dict[Any, Any]) -> Browser:
 @pytest.fixture(scope="session")
 def browser(launch_browser: Callable[[], Browser]) -> Generator[Browser, None, None]:
     browser = launch_browser()
-    yield browser
-    browser.close()
+    try:
```
Comment (kumaraditya303) [inline pytest_playwright/pytest_playwright.py:124]: Without this in case of an exception, browser keeps running in background

**Comment by mxschmitt** [comment]
I don't see a reason to implement that since it seems no user yet ran into that issue and in the referred issue they just guess it could happen potentially.
(Pytest selenium e.g. does it also not)

**Comment by JoelEinbinder** [comment]
@pavelfeldman this looks reasonable to me but I don't know python.

**Comment by mxschmitt** [comment]
I tested it with the page fixture in the following way:

```py
@pytest.fixture
def page(context: BrowserContext, base_url: str) -> Generator[Page, None, None]:
    page = context.newPage()
    try:
        yield page
        print("normal")
    except Exception:
        print("exception")
    page.close()
```

and `normal` got logged, so I think the proposed changes won't improve the user experience since the browser get automatically closed already after the yield / test run even when an exception got thrown.

**Comment by pavelfeldman** [comment]
My understanding is that this is a noop since yielding in pytest never throws. Could you please re-file it against playwright-python with a test that fails without this change?

**Pull request closed without merging**